### ISSUE TITLE
InvalidArgumentError (see above for traceback): Default AvgPoolingOp only supports NHWC. how to solve this issue?

### ISSUE THREAD

**Issue by hsl0529** [melodyguan/enas#12]
Caused by op u'child/layer_3/pool_at_3/from_0/AvgPool_1', defined at:
  File "main.py", line 359, in <module>
    tf.app.run()
  File "/home/sileihe/anaconda3/envs/test_py2/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "main.py", line 355, in main
    train()
  File "main.py", line 223, in train
    ops = get_ops(images, labels)
  File "main.py", line 171, in get_ops
    child_model.connect_controller(controller_model)
  File "/home/sileihe/enas-master/src/cifar10/general_child.py", line 705, in connect_controller
    self._build_train()
  File "/home/sileihe/enas-master/src/cifar10/general_child.py", line 595, in _build_train
    logits = self._model(self.x_train, is_training=True)
  File "/home/sileihe/enas-master/src/cifar10/general_child.py", line 222, in _model
    layer, out_filters, 2, is_training)
  File "/home/sileihe/enas-master/src/cifar10/general_child.py", line 165, in _factorized_reduction
    path2, [1, 1, 1, 1], stride_spec, "VALID", data_format=self.data_format)
  File "/home/sileihe/anaconda3/envs/test_py2/lib/python2.7/site-packages/tensorflow/python/ops/nn_ops.py", line 1930, in avg_pool
    name=name)
  File "/home/sileihe/anaconda3/envs/test_py2/lib/python2.7/site-packages/tensorflow/python/ops/gen_nn_ops.py", line 68, in _avg_pool
    data_format=data_format, name=name)
  File "/home/sileihe/anaconda3/envs/test_py2/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/sileihe/anaconda3/envs/test_py2/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 2956, in create_op
    op_def=op_def)
  File "/home/sileihe/anaconda3/envs/test_py2/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Default AvgPoolingOp only supports NHWC.
	 [[Node: child/layer_3/pool_at_3/from_0/AvgPool_1 = AvgPool[T=DT_FLOAT, data_format="NCHW", ksize=[1, 1, 1, 1], padding="VALID", strides=[1, 1, 2, 2], _device="/job:localhost/replica:0/task:0/device:CPU:0"](child/layer_3/pool_at_3/from_0/strided_slice)]]

**Comment by hyhieu** [comment]
We believe this is an issue of different TensorFlow versions.
